frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -130,61 +167,227 @@ const ShimmeringBody: React.FC<ShimmeringBodyProps> = ({
+      rowKey={columnKey}
+      expandedRows={expandedRows}
+      onExpand={onExpand}
+      onCollapse={onCollapse}
+      rowValues={rowValues}
+      onClick={setExpandedRows}
+      isSelectedRow={currentSelectedKey === columnKey}
+      nestedLevel={nestedLevel}
+    />
+  );
+
+  return (
+    <React.Fragment key={columnKey}>
+      <tr
+        className={`ams-table-row ${
+          rowValues.isNestedColumn && 'is-nested-column-row'
+        } ${currentSelectedKey === columnKey && 'is-selected-row'}`}
+        key={columnKey}
+        style={rowStyles}
+        ref={expandRowRef}
+      >
+        <>
+          {Object.entries(rowValues)
+            .filter(([key]) => fields.includes(key))
+            .map(([key, value], rowIndex) => {
+              const columnInfo = columns.find(({ field }) => field === key);
+              const horAlign: TextAlignmentValues = columnInfo
+                ? columnInfo.horAlign || DEFAULT_TEXT_ALIGNMENT
+                : DEFAULT_TEXT_ALIGNMENT;
+              const width =
+                columnInfo && columnInfo.width
+                  ? `${columnInfo.width}px`
+                  : DEFAULT_CELL_WIDTH;
+              // TODO: Improve the typing of this
+              let cellContent: React.ReactNode | typeof value = value;
+              if (columnInfo && columnInfo.component) {
+                cellContent = columnInfo.component(value, rowIndex, rowValues);
+              }
+
+              const isFirstCell =
+                fields.findIndex((field) => field === key) === 0;
+              const hasExpandingButton = isFirstCell && rowValues.isExpandable;
+
+              let cellStyle;
+              if (isFirstCell) {
+                cellStyle = {
+                  width,
+                  paddingLeft: getIndentationPaddingSize(
+                    nestedLevel,
+                    rowValues.isExpandable
+                  ),
+                };
+              } else {
+                cellStyle = { width };
+              }
+
+              return (
+                <td
+                  className={`ams-table-cell ${getCellAlignmentClass(
+                    horAlign
+                  )}`}
+                  key={`index:${rowIndex}`}
+                  style={cellStyle}
+                >
+                  <span
+                    className={`${
+                      isFirstCell && 'ams-table-first-cell-contents'
+                    }`}
+                  >
+                    {hasExpandingButton && expandingButton}
+                    {cellContent}
+                  </span>
+                </td>
+              );
+            })}
+        </>
+      </tr>
+    </React.Fragment>
+  );
+};
+
+type RowKey = string;
+
+const getTableRows = (
+  data,
+  columns,
+  currentSelectedKey,
+  preExpandPanelKey,
+  rowStyles,
+  onExpand,
+  onCollapse,
+  expandRowRef,
+  expandedRows,
+  setExpandedRows,
+  formatChildrenData,
+  preExpandRightPanel,
+  nestedLevel
+) =>
```

Comment:
Let's change the input to use an object instead of this long list of arguments. This is better because it removes the need of the exact order.